client/me/purchases/manage-purchase/index.jsx

```diff
@@ -450,7 +477,7 @@ const ManagePurchase = React.createClass( {
 		const purchase = getPurchase( this.props ),
 			{ domain, id } = purchase;
 
-		if ( isExpired( purchase ) || ! isCancelable( purchase ) ) {
+		if ( ! isCancelable( purchase ) ) {
```

Comment:
I moved `isExpired` inside `isCancelable`, because you can't cancel expired product.